### PR TITLE
chore: fix circleci publish-workflow for merging and publishing (@W-8190683)

### PR DESCRIPTION
**Pull request by sfsholden**
### What does this PR do?
1. Adds RELEASE_VERSION logic for publish-workflow.
2. Fixes post-publish merge logic.

### What issues does this PR fix or reference?
@W-8190683@

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -412,6 +412,7 @@ jobs:
           message: 'Starting publish for VSCode Extensions'
       - ssh-config
       - checkout
+      - prepare-release-environment-variables
```
Comment (sfsholden) [inline .circleci/config.yml:415]: In the previous release for v49.13.0 there was an issue caused by the previous refactor to prepare-gus-environment-variables. The logic for determining the RELEASE_VERSION was moved into its own command, but wasn't added here for the publish-workflow.

This had two effects:
1. The tag for 49.13.0 wasn't added correctly.
2. The commit message for 'Updated SHA256 v' was missing the version.

Did this have a negative impact on the release itself? - No

File: .circleci/config.yml [hunk 2]
```diff
@@ -446,6 +447,7 @@ jobs:
           name: 'Merge changes from main back into develop'
           command: |
             git checkout develop
+            git pull
```
[the remark this answers is not in the document]
Comment (lcampos) [reply .circleci/config.yml:450]: That's weird, I would assume the `- checkout` task on line 414 would take care of this.

[the remark this answers is not in the document]
Comment (sfsholden) [reply .circleci/config.yml:450]: I would assume so as well. From testing in my forked repo, this seems to resolve the issue.

[the remark this answers is not in the document]
Comment (lcampos) [reply .circleci/config.yml:450]: Ok, maybe there's something else missing in our configs. Let's add this to the list of things to ask next time we chat with CircleCI

**Review by lcampos — APPROVED** [review]

**Pull request merged**